slash/core/result.py

```diff
@@ -35,6 +35,8 @@ def __init__(self, test_metadata=None):
         self.test_metadata = test_metadata
         #: dictionary to be use by tests and plugins to store result-related information for later analysis
         self.data = {}
+        self._start_time = datetime.now()  # in case `mark_started` or `mark_finished` we're not called
+        self._end_time = self._start_time  # the computed value of duration should be 0
```

Comment:
I think Duration should be `None` for non-started results, and for non-ended result should be either `None` or the actual time that elapsed.

Reply:
done